pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/when/impl/ExpressionConditional.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.pipeline.modeldefinition.when.impl;
+
+import hudson.AbortException;
+import hudson.EnvVars;
+import hudson.Extension;
+import org.apache.tools.ant.types.selectors.SelectorUtils;
+import org.jenkinsci.Symbol;
+import org.jenkinsci.plugins.pipeline.modeldefinition.model.StepsBlock;
+import org.jenkinsci.plugins.pipeline.modeldefinition.when.DeclarativeStageConditional;
+import org.jenkinsci.plugins.pipeline.modeldefinition.when.DeclarativeStageConditionalDescriptor;
+import org.jenkinsci.plugins.workflow.cps.CpsScript;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.apache.commons.lang.StringUtils.isEmpty;
+
+/**
+ * Stage condition based on the current branch. i.e. the env var BRANCH_NAME.
+ * As populated by {@link jenkins.branch.BranchNameContributor}
+ */
+public class ExpressionConditional extends DeclarativeStageConditional<ExpressionConditional> {
+    private final StepsBlock block;
+
+    @DataBoundConstructor
+    public ExpressionConditional(StepsBlock block) {
+        this.block = block;
+    }
+
+    public StepsBlock getBlock() {
+        return block;
+    }
+
+    public static boolean booleanFromReturn(Object o) throws AbortException {
```

Comment:
Well I don't think we need to make sure that it actually is a boolean, just that Groovy can evaluate it to `true`/`false` and would be much more convenient if we let groovy handle that then trying to emulate it in Java imho it is much more convenient to just do `{ return env.MY_ENV }` instead of `{return env.My_ENV != null && !env.MY_ENV,isEmpty()}`